### PR TITLE
64 list packages versions and network compatibility

### PR DESCRIPTION
**Pull request by Ben-Rey**
This is a first proposition on how to display versions

Version are stored in a json located in a github gist. For test purpose I created one on my account : https://gist.github.com/Ben-Rey/9f475f1c5e8e0f78fa1570a727eef344

Why did we use a json ?
- This json will allow us to centralise versions informations in a single place.
- We could directly update this json (so by extension the doc) using ce CI when releasing a new project
- The JSON file can serve as a reliable data source for test repositories responsible for testing projects across different network

I keep the design simple but can be improved

### CODE REVIEW

**Comment by gregLibert** [comment]
@ben I have the feeling that this information should be in the Networks & Faucets (build section). Hence you will be able to have everything you need to connect to a network at the same place.

We can also duplicate the information to multiple pages if the source is the same :-)

@qdrn

File: src/components/versions-table.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    return `${project.install}@${compatibility.version}`;
+  };
+
+  if (!data) {
+    return <div></div>;
+  }
+
+  return (
+    <div style={{ marginTop: "50px" }}>
+      <div>
```
Comment (gregLibert) [inline src/components/versions-table.tsx:47]: You should remove useless div:
```suggestion
  <>
```

File: src/components/versions-table.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+              <th>Install (Buildnet)</th>
+            </tr>
+          </thead>
+          <tbody>
+            {data.projects.map((project, i) => (
+              <tr key={i}>
+                <td>{project.title}</td>
+                {project.compatibilities.map((compatibilities) => (
+                  <td>{compatibilities.version}</td>
+                ))}
```
Comment (gregLibert) [inline src/components/versions-table.tsx:64]: int the ma it's a compatibility, not compatibilities:
```suggestion
              <td>{project.title}</td>
              {project.compatibilities.map((comp) => (
                <td key={comp.network}>{comp.version}</td>
              ))}
```

File: src/components/versions-table.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                <td
+                  onClick={() => handleCopy(project.install)}
+                  style={{ cursor: "pointer" }}
+                >
+                  {getInstall(project, "buildnet")}
+                </td>
```
Comment (gregLibert) [inline src/components/versions-table.tsx:70]: You should use a code bloc isntead:
```suggestion
              <td>
                <CodeBlock className="language-bash">
                  {getInstall(project, "buildnet")}
                </CodeBlock>
              </td>
```

File: src/components/versions-table.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+              </tr>
+            ))}
+          </tbody>
+        </table>
+      </div>
+    </div>
```
Comment (gregLibert) [inline src/components/versions-table.tsx:76]: Remove useless trailing div:
```suggestion
    </>
```

File: src/utils/helpers.tsx [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const handleCopy = async (text) => {
+  navigator.clipboard.writeText(text);
+};
```
Comment (gregLibert) [inline src/utils/helpers.tsx:3]: useless

File: src/components/versions-table.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  "https://gist.githubusercontent.com/Ben-Rey/9f475f1c5e8e0f78fa1570a727eef344/raw/gistfile1.txt";
+
+export default function VersionsTable() {
+  const [data, setData] = useState<CompatibilityData>();
+
+  useEffect(() => {
+    fetch(GIST_URL)
+      .then((response) => response.json())
+      .then((data) => setData(data));
+  }, []);
```
[the remark this answers is not in the document]
Comment (Ben-Rey) [reply src/components/versions-table.tsx]: Done

File: src/components/versions-table.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+  const getInstall = (project: Project, network: string) => {
+    const compatibility: Compatibility = project.compatibilities.find(
+      (compatibility) => compatibility.network === network
+    );
+    return `${project.install}@${compatibility.version}`;
```
[the remark this answers is not in the document]
Comment (Ben-Rey) [reply src/components/versions-table.tsx]: Done

File: src/components/versions-table.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  };
+
+  if (!data) {
+    return <div></div>;
```
[the remark this answers is not in the document]
Comment (Ben-Rey) [reply src/components/versions-table.tsx]: done

File: src/components/versions-table.tsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+interface CompatibilityData {
+  title: string;
+  projects: Project[];
+  networks: string[];
+}
+
+const GIST_URL =
+  "https://gist.githubusercontent.com/Ben-Rey/9f475f1c5e8e0f78fa1570a727eef344/raw/gistfile1.txt";
```
[the remark this answers is not in the document]
Comment (Ben-Rey) [reply src/components/versions-table.tsx:23]: Sorry, not sure to get the question. 
The JSON should be stored in a gist in massa organisation.
The link should stay the same so I can put it in a constants file

**Comment by qdrn** [comment]
> @ben I have the feeling that this information should be in the Networks & Faucets (build section). Hence you will be able to have everything you need to connect to a network at the same place.
> 
> We can also duplicate the information to multiple pages if the source is the same :-)
> 
> @qdrn wdyt?

Yes, I think it makes more sense to put it in the network page and to put a note on the SDK page that link that links to it (or duplicate if you think it's better).

Overall I like the proposal that you made of having this json file that can be automatically updated. Much less error prone and much more convenient once setup!

Design is simple but good enough imo. We can have someone with more expertise go through all the designs once we are done with a first version (and done with more urgent matters!).

Congrats!

File: src/components/versions-table.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (gregLibert) [reply src/components/versions-table.tsx]: But right now the gist file 1 is in your personal account, right ?

[the remark this answers is not in the document]
Comment (Ben-Rey) [reply src/components/versions-table.tsx]: Yes we need to create one with the massa account, sorry if I forgot to mention it. Then we'll update the URL.
Who can create a gist in massa organisation ?

**Review by gregLibert — APPROVED** [review]
Sounds good to me. Except for the fact that the piece of information should also be present in the network and faucet section

**Comment by qdrn** [comment]
I think it's better like this, thanks!

**Review by qdrn — CHANGES_REQUESTED** [review]
Let's put the json in a file in this project that will be updated by the CI of the different projects as discussed.

**Comment by Ben-Rey** [comment]
> Let's put the json in a file in this project that will be updated by the CI of the different projects as discussed.

I need to put this pr on hold for 1 day or 2 because we found a way to use tags so the developer will just have to do npm install massalabs/massa-web3@buildnet or massalabs/massa-web3@testnet or massalabs/massa-web3@latest ...
I just need the time to implement it on each repo

**Comment by Ben-Rey** [comment]
> > Let's put the json in a file in this project that will be updated by the CI of the different projects as discussed.
> 
> I need to put this pr on hold for 1 day or 2 because we found a way to use tags so the developer will just have to do npm install massalabs/massa-web3@buildnet or massalabs/massa-web3@testnet or massalabs/massa-web3@latest ... I just need the time to implement it on each repo



> Let's put the json in a file in this project that will be updated by the CI of the different projects as discussed.

@julienbrs will check all versions to make sure we're providing the right information, then I'll put in the JSON.

**Comment by qdrn** [comment]
Yeah no problem just keeping track! Thanks for the info.

File: docs/build/networks-faucets.mdx [hunk 10]
```diff
@@ -91,3 +116,9 @@ Sure! You'll need:
 
 - A Massa wallet (Don't have one yet? Generate addresses on the [Explorer website](https://massa.net/testnet/wallet))
 - A [Discord account](https://support.discord.com/hc/en-us/articles/360033931551-Getting-Started)
+
+## Library Compatibility with Massa Networks
+
+The following table outlines library compatibility across the Massa networks. Use this as a guide to select the appropriate library based on the network you're working with and your project needs.
+
+<VersionsTable />
```
Comment (qdrn) [inline docs/build/networks-faucets.mdx:124]: There's an issue here. See CI also.

File: docs/build/sdk.mdx [hunk 11]
```diff
@@ -3,4 +3,18 @@ id: sdk
...
+
+Take a look at the table below to identify the right version for your desired network.
+
+This table provides a concise overview of the different versions of our projects that are compatible with the different Networks.
+
+We offer distinct versions for each network type to ensure optimal performance and compatibility. The table also includes the respective installation commands for each project.
+
+Take a look at the table below to identify the right version for your desired network.
+
+<VersionsTable />
```
Comment (qdrn) [inline docs/build/sdk.mdx:20]: Same here.

**Review by qdrn — CHANGES_REQUESTED** [review]

**Review by qdrn — APPROVED** [review]

**Pull request merged**